cocos/core/game.ts

```diff
@@ -642,32 +651,26 @@ export class Game extends EventTarget {
         this._lastTime = performance.now();
         const frameRate = this.config.frameRate;
         this._frameTime = 1000 / frameRate;
-        legacyCC.director._maxParticleDeltaTime = this._frameTime / 1000 * 2;
         if (JSB || RUNTIME_BASED) {
             // @ts-ignore
             jsb.setPreferredFramesPerSecond(frameRate);
-            window.rAF = window.requestAnimationFrame;
-            window.cAF = window.cancelAnimationFrame;
-        }
-        else {
-            if (this._intervalId) {
-                window.cAF(this._intervalId);
-                this._intervalId = 0;
-            }
-
-            const rAF = window.requestAnimationFrame = window.requestAnimationFrame ||
-            window.webkitRequestAnimationFrame ||
-            window.mozRequestAnimationFrame ||
-            window.oRequestAnimationFrame ||
-            window.msRequestAnimationFrame;
+            this.requestAnimationFrame = window.requestAnimationFrame;
+            this.cancelAnimationFrame = window.cancelAnimationFrame;
+        } else {
+            this._dropFrame();
+
+            const rAF =
+                (window.requestAnimationFrame ||
+                window.webkitRequestAnimationFrame ||
+                window.mozRequestAnimationFrame ||
+                window.oRequestAnimationFrame ||
+                window.msRequestAnimationFrame) as (undefined | typeof requestAnimationFrame);
```

Comment:
@holycanvas  Notable change here: replace `// @ts-ingore` with a type cast.

Reply:
I guess some external contributors are less likely to change the existing structure of the engine to avoid introducing some other problems when modifying the engine, so they would prefer to copy the code